HPXMLtoOpenStudio/resources/hvac.rb

```diff
@@ -1375,6 +1385,29 @@ def self.apply_setpoints(model, runner, weather, hvac_control, living_zone)
     end
     clg_weekend_setpoints = clg_weekday_setpoints
 
+    # Optionally apply 24-hr cooling setpoint schedules
+    if (not hvac_control.weekday_cooling_setpoints.nil?) && (not hvac_control.weekend_cooling_setpoints.nil?)
+      clg_weekday_setpoints = hvac_control.weekday_cooling_setpoints.split(', ').map { |i| i.to_f }
+      clg_weekday_setpoints = [clg_weekday_setpoints] * 12
+      if not clg_ceiling_fan_offset.nil?
+        HVAC.get_default_ceiling_fan_months(weather).each_with_index do |operation, m|
+          next unless operation == 1
+
+          clg_weekday_setpoints[m] = [clg_weekday_setpoints[m], Array.new(24, clg_ceiling_fan_offset)].transpose.map { |i| i.reduce(:+) }
+        end
+      end
```

Comment:
Duplicated code (and twice at that! here and below). 👎 

Please refactor the code so that the ceiling fan offset logic only needs to be applied once.

Reply:
Good catch.